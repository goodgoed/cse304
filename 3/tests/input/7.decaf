class WithConstructor {
  int x;

  WithConstructor(int x) {
    this.x = x;
  }
}