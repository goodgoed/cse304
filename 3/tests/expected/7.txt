--------------------------------------------------------------------------
Class Name: WithConstructor
Superclass Name: 
Fields:
FIELD: 1, x, WithConstructor, private, instance, int
Constructors:
CONSTRUCTOR: 1, private
Constructor Parameters: 1
Variable Table:
VARIABLE: 1, x, formal, int
Constructor Body:
Block([
Expr(Assign(Field-access(This, x), Variable(1)))
])
Methods:
--------------------------------------------------------------------------
